### PR TITLE
Fixed Image collection query

### PR DESCRIPTION
**Pull request by aagm**
I have fixed the query for one where condition and an issue with alias not working on one aggregation.

### CODE REVIEW

File: tests/test_imageCollection_func.py [hunk 1]
```diff
@@ -68,3 +68,9 @@ def test_image_collection_group_by_query():
     response = SQL2GEE(JsonSql(sql).to_json(), geojson=r).response()
     assert len(response) == 10, "BASIC limit query incorrect"
     return
+
+def test_image_collection_group_by_with_geometry_where_condition():
+    sql = "select system:index, first(q25) as q25,first(q50) as q50, first(q75) as q75, mean(q75) as q75_mean from 'projects/resource-watch-gee/cli_051_nexgddp_annual_tasmin' where ST_INTERSECTS(ST_SetSRID(ST_GeomFromGeoJSON('{\"type\":\"Point\",\"coordinates\":[-110.22939224192194,19.986126139624318]}'),4326),the_geom) group by system:index"
```
Comment (tiagojsag) [inline tests/test_imageCollection_func.py]: I'd prefer to have a test that runs on top of a public dataset. Running tests on a "private" dataset basically ties the tests to a single GCP account.

File: test.py [hunk 2]
```diff
@@ -1,9 +1,13 @@
-import ee;
```
Comment (tiagojsag) [inline test.py]: AFAICT this whole file is not necessary. can it be removed?

File: sql2gee/collection.py [hunk 3]
```diff
@@ -37,10 +37,12 @@ def _where(self):
...
         """
-        if 'where' in self._parsed and self._parsed['where']:
-            if self.geometry:
+        if 'where' in self._parsed and len(self._parsed['where']) > 0:
+            if self.geometry and 'filter' in self._filters:
                 self._asset = self._asset.filter(self._filters['filter']).filterBounds(self.geometry)
-            else:
+            elif self.geometry:
+                self._asset = self._asset.filterBounds(self.geometry)
+            elif 'filter' in self._filters:
```
Comment (tiagojsag) [inline sql2gee/collection.py:45]: what happens when `if not self.geometry and 'filter' not in self._filters` ?

File: sql2gee/collection.py [hunk 4]
```diff
@@ -100,14 +102,21 @@ def _output(self):
         return _Output
 
     def _mapOutputIList(self, img):
-        if len(self._output['alias']['result']) > 0:
+        """
+        """
+        # @debt: inside the reducers we will need to use setOutputs
+        if len(self._output['alias']['result']) > 0 and type(img) is dict:
```
Comment (tiagojsag) [inline sql2gee/collection.py:112]: is there any scenario where `type(img) is not dict` ? 
At least from the existing tests, it seems it's always a dictionary, so I'm wondering if this is redundant

File: Makefile [hunk 5]
```diff
@@ -3,14 +3,24 @@
 
 PYTHON=`which python`
 
+# cli prefix for commands to run in container
```
Comment (tiagojsag) [inline Makefile]: personally I would remove the makefile, and replace it with a proper README and/or a shell script.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.8.6 as user
```
Comment (tiagojsag) [inline Dockerfile]: Let's not add a docker dependency that future contributors have to maintain.

File: README.md [hunk 7]
```diff
@@ -47,6 +47,13 @@ tox
 
 ## Development
 
+To develop it with docker and docker compose,  execute:
```
Comment (tiagojsag) [inline README.md]: again, docker dependency creep here.

File: tests/test_imageCollection_func.py [hunk 8]
```diff
@@ -68,3 +68,9 @@ def test_image_collection_group_by_query():
...
+    response = SQL2GEE(JsonSql(sql).to_json()).response()
+    assert len(response) > 0, f"Empty response: {response}"
```
Comment (tiagojsag) [inline tests/test_imageCollection_func.py]: Testing that a response is not empty does not validate that the query is returning what it should, it just ensures it's not returning one of the things that it shouldn't. Please make these tests more restrictive, so we can have stronger confidence in the results.

File: .env.sample [hunk 9]
```diff
@@ -0,0 +1,3 @@
+EE_ACCOUNT=<my-account@gmail.com>
```
Comment (tiagojsag) [inline .env.sample]: I think this should not be here

**Review by tiagojsag — CHANGES_REQUESTED** [review]
Seems like there are a few things to improve here, but overall looks like a move in the right direction :)

File: tests/test_imageCollection_func.py (hunk 1 above)
Comment (aagm) [reply tests/test_imageCollection_func.py]: Not really if it is public read but I can change the dataset to one in Google public dataset if you will feel more confortable ❤️

File: test.py (hunk 2 above)
Comment (aagm) [reply test.py]: I do use it for test. I can add it to git ignore if that suits you better ❤️

File: sql2gee/collection.py (hunk 3 above)
Comment (aagm) [reply sql2gee/collection.py:45]: Nothing will be added which should be correct

File: sql2gee/collection.py (hunk 4 above)
Comment (aagm) [reply sql2gee/collection.py:112]: Is a pain in the ass the difference to type it in gee ee.Dictionary makes the operation happend on their server and not in ours

File: tests/test_imageCollection_func.py (hunk 1 above)
Comment (tiagojsag) [reply tests/test_imageCollection_func.py]: yes, please. So one of the issues we had recently is that we were using our own datasets to test, and that disappeared for whatever reason, so I had to ask you for new ones. "public" datasets do come with their own risks, of changes we dont control, but I still feel like that's a lesser risk

File: test.py (hunk 2 above)
Comment (tiagojsag) [reply test.py]: I'm just trying to understand what it's for. Pytest + tox seem to do the job for me, and less is more, so I'm trying to understand how this is used, and what's the best way to achieve that goal.

File: sql2gee/collection.py (hunk 4 above)
Comment (tiagojsag) [reply sql2gee/collection.py:112]: ok, but does it ever happen that img is something other than a dict?

**Pull request closed without merging**